internal/sync/lifecycle.go

```diff
@@ -115,12 +155,15 @@ func (l *LifecycleOnce) Stop(f func() error) error {
 		return l.stopErr
 	}
 
-	if f == nil {
-		f = func() error { return nil }
+	// Unblock WaitForStart callers (ccounting Stop() without Start())
+	if l.started != nil {
+		close(l.started)
```

Comment:
this could panic, if we attempt to close an already closed channel go will panic.
https://play.golang.org/p/8f3QCd0goD

We should probably have another channel for stopped.